src/overlays/actors/ovl_En_Skb/z_en_skb.c

```diff
@@ -66,55 +74,499 @@ static ColliderJntSphInit D_80AFE068 = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skb/func_80AFD540.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skb/func_80AFD59C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skb/func_80AFD644.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skb/func_80AFD6CC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skb/func_80AFD7B4.s")
+static DamageTable D_80AFE078 = {
+    /* Deku nut      */ DMG_ENTRY(0 , 0x1),
+    /* Deku stick    */ DMG_ENTRY(2 , 0xF),
+    /* Slingshot     */ DMG_ENTRY(1 , 0xF),
+    /* Explosive     */ DMG_ENTRY(2 , 0xF),
+    /* Boomerang     */ DMG_ENTRY(0 , 0x1),
+    /* Normal arrow  */ DMG_ENTRY(2 , 0xF),
+    /* Hammer swing  */ DMG_ENTRY(2 , 0xF),
+    /* Hookshot      */ DMG_ENTRY(0 , 0x1),
+    /* Kokiri sword  */ DMG_ENTRY(1 , 0xE),
+    /* Master sword  */ DMG_ENTRY(2 , 0xF),
+    /* Giant's Knife */ DMG_ENTRY(4 , 0xF),
+    /* Fire arrow    */ DMG_ENTRY(4 , 0x7),
+    /* Ice arrow     */ DMG_ENTRY(2 , 0xF),
+    /* Light arrow   */ DMG_ENTRY(2 , 0xF),
+    /* Unk arrow 1   */ DMG_ENTRY(2 , 0xF),
+    /* Unk arrow 2   */ DMG_ENTRY(0 , 0x0),
+    /* Unk arrow 3   */ DMG_ENTRY(0 , 0x0),
+    /* Fire magic    */ DMG_ENTRY(4 , 0x7),
+    /* Ice magic     */ DMG_ENTRY(0 , 0x6),
+    /* Light magic   */ DMG_ENTRY(3 , 0xD),
+    /* Shield        */ DMG_ENTRY(0 , 0x0),
+    /* Mirror Ray    */ DMG_ENTRY(0 , 0x0),
+    /* Kokiri spin   */ DMG_ENTRY(1 , 0xD),
+    /* Giant spin    */ DMG_ENTRY(4 , 0xF),
+    /* Master spin   */ DMG_ENTRY(2 , 0xF),
+    /* Kokiri jump   */ DMG_ENTRY(2 , 0xF),
+    /* Giant jump    */ DMG_ENTRY(8 , 0xF),
+    /* Master jump   */ DMG_ENTRY(4 , 0xF),
+    /* Unknown 1     */ DMG_ENTRY(0 , 0x0),
+    /* Unblockable   */ DMG_ENTRY(0 , 0x0),
+    /* Hammer jump   */ DMG_ENTRY(4 , 0xF),
+    /* Unknown 2     */ DMG_ENTRY(0 , 0x0),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skb/func_80AFD880.s")
+const ActorInit En_Skb_InitVars = {
+    ACTOR_EN_SKB,
+    ACTORCAT_ENEMY,
+    FLAGS,
+    OBJECT_SKB,
+    sizeof(EnSkb),
+    (ActorFunc)EnSkb_Init,
+    (ActorFunc)EnSkb_Destroy,
+    (ActorFunc)EnSkb_Update,
+    (ActorFunc)EnSkb_Draw,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skb/func_80AFD968.s")
+Vec3f D_80AFE0B8 = { 0.0f, 8.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skb/EnSkb_Update.s")
+Vec3f D_80AFE0C4 = { 0.0f, -1.5f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skb/func_80AFDD30.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(targetArrowOffset, 2000, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(gravity, -2000, ICHAIN_STOP),
+};
 
+void EnSkb_SetupAction(EnSkb* this, EnSkbActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnSkb_SpawnDebris(GlobalContext* globalCtx, EnSkb* this, Vec3f* spawnPos) {
+    Vec3f pos;
+    Vec3f vel = D_80AFE0B8;
```

Comment:
Can these be inlined data?

Reply:
Not that I know of? When I inlined the data the offset was too much:
```C
20:    addiu   t7,t7,%lo(sVelocity)               i 20:    addiu   t7,t7,%lo(.data+0xa0)
```

Reply:
Did you try it with make or -mw? Also it might have to be non static.